Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CorrelationAttributeUtil.java

```diff
@@ -215,7 +215,14 @@ private static void makeCorrAttrFromAcctArtifact(List<CorrelationAttributeInstan
         if (Account.Type.DEVICE.getTypeName().equalsIgnoreCase(accountTypeStr) == false) {
 
             // Get the corresponding CentralRepoAccountType from the database.
-            CentralRepoAccountType crAccountType = CentralRepository.getInstance().getAccountTypeByName(accountTypeStr);
+            CentralRepoAccountType crAccountType;
+            try {
+                // @@TODO Vik-6136: CR currently does not know of custom account types.  
+                // This try/catch prevents unnecessary exceptions, till CR handles implement custom account types
+                crAccountType = CentralRepository.getInstance().getAccountTypeByName(accountTypeStr);
+            } catch (CentralRepoException ex) {
+                return;
```

Comment:
That is exactly how I started out,  till I realized there is no easy, inexpensive and definitive way to check if the account we have is one of the default account types, so it might just be better to ignore the exception this way.

The only way to check is match on the DisplayName of the CorrelationAttrribute.Type from the list returned by CorrelationAttributeInstance.getDefaultCorrelationTypes().  In theory, the display names don't have to be unique - so there is a possibility of false positive - but since we set them up ourselves, I suppose we can assume they are unique.

And getting the display name of the account type, from the account type that we have (from the TSK_ACCOUNT_TYPE attribute value of the artifact) in itself requires a roundtrip to the Case DB:
 skCase. getCommunicationsManager().getAccountType(acountTypeName)

It seems we would be adding more CPU cycles to EVERY call of makeCorrAttrFromAcctArtifact(),  to prevent an error message that has a low probability of happening.

If you still prefer this solution though. I'm happy to change it.

Reply:
I don't understand why this is not simpler. In CorrelationAttributeUtil.makeCorrAttrFromAcctArtifact we have a BlackboardArtifact of type TSK_ACCOUNT in hand. We have a BlackboardAttribute of type TSK_ACCOUNT_TYPE from that artifact. If the string value of the attribute is is not equal to the type name of one of the account types in Account.PREDEFINED_ACCOUNT_TYPES, then we are done. What am I missing?

Reply:
(SMH) Ahh, I see it now. You're saying to search the account type at hand with the Account.PREDEFINED_ACCOUNT_TYPES list.

I was thinking of searching for the corresponding Correlation Attribute instance name, in the CorrelationAttributeInstance.getDefaultCorrelationTypes().

I'll fix it.  Thanks.